### ISSUE TITLE
GStreamer broke GStreamer Java

### ISSUE THREAD

**Issue by sblantipodi** [gstreamer-java/gst1-java-core#286]
As title.

I've been using GStreamer Java for my Java [Luciferin project](https://github.com/sblantipodi/firefly_luciferin) for quite some time now, and I knew this moment would eventually come.
It seems that GStreamer has broken compatibility with GStreamer Java in recent commits.

Unfortunately, I'm not an expert in GStreamer, nor in this binding, so my contribution here is rather limited.

I noticed this issue because I'm currently waiting for new features in version 1.26 and I've been trying the [1.25 nightly builds from time to time](https://gitlab.freedesktop.org/gstreamer/cerbero/-/pipelines?scope=finished&page=1&ref=main&source=schedule)...

**Until 2 months ago, the nightly builds produced installers that worked perfectly with GStreamer Java.**

Now I've noticed that the latest commits have broken something because as soon as I try to start my program with the latest nightly build I get this stack trace.

The error happens on the
Gst.init("myapp", "");
function

and in particular here:
```
if (!GstAPI.GST_API.gst_init_check(argv.argcRef, argv.argvRef, error)) {
  INIT_COUNT.decrementAndGet();
  throw new GstException(extractError(error[0]));
}
```

```
Exception in Application start method
Exception in thread "main" java.lang.RuntimeException: Exception in Application start method
    at [javafx.graphics@23/com.sun.javafx.application.LauncherImpl.launchApplication1](mailto:javafx.graphics@23/com.sun.javafx.application.LauncherImpl.launchApplication1)(LauncherImpl.java:893)
    at [javafx.graphics@23/com.sun.javafx.application.LauncherImpl.lambda$launchApplication$2](mailto:javafx.graphics@23/com.sun.javafx.application.LauncherImpl.lambda$launchApplication$2)(LauncherImpl.java:196)
    at java.base/java.lang.Thread.run(Thread.java:1575)
Caused by: java.lang.Error: Invalid memory access
    at [com.sun.jna@5.15.0/com.sun.jna.Native.invokeInt](mailto:com.sun.jna@5.15.0/com.sun.jna.Native.invokeInt)(Native Method)
    at [com.sun.jna@5.15.0/com.sun.jna.Function.invoke](mailto:com.sun.jna@5.15.0/com.sun.jna.Function.invoke)(Function.java:429)
    at [com.sun.jna@5.15.0/com.sun.jna.Function.invoke](mailto:com.sun.jna@5.15.0/com.sun.jna.Function.invoke)(Function.java:364)
    at [com.sun.jna@5.15.0/com.sun.jna.Library$Handler.invoke](mailto:com.sun.jna@5.15.0/com.sun.jna.Library$Handler.invoke)(Library.java:270)
    at jdk.proxy1/jdk.proxy1.$Proxy17.gst_init_check(Unknown Source)
    at [org.freedesktop.gstreamer@1.4.0/org.freedesktop.gstreamer.Gst.init](mailto:org.freedesktop.gstreamer@1.4.0/org.freedesktop.gstreamer.Gst.init)(Gst.java:502)
    at [org.freedesktop.gstreamer@1.4.0/org.freedesktop.gstreamer.Gst.init](mailto:org.freedesktop.gstreamer@1.4.0/org.freedesktop.gstreamer.Gst.init)(Gst.java:453)
    at org.dpsoftware/org.dpsoftware.grabber.GrabberManager.launchAdvancedGrabber(GrabberManager.java:104)
    at org.dpsoftware/org.dpsoftware.FireflyLuciferin.launchGrabberAndConsumers(FireflyLuciferin.java:330)
    at org.dpsoftware/org.dpsoftware.FireflyLuciferin.start(FireflyLuciferin.java:286)
    at [javafx.graphics@23/com.sun.javafx.application.LauncherImpl.lambda$launchApplication1$9](mailto:javafx.graphics@23/com.sun.javafx.application.LauncherImpl.lambda$launchApplication1$9)(LauncherImpl.java:839)
    at [javafx.graphics@23/com.sun.javafx.application.PlatformImpl.lambda$runAndWait$12](mailto:javafx.graphics@23/com.sun.javafx.application.PlatformImpl.lambda$runAndWait$12)(PlatformImpl.java:483)
    at [javafx.graphics@23/com.sun.javafx.application.PlatformImpl.lambda$runLater$10](mailto:javafx.graphics@23/com.sun.javafx.application.PlatformImpl.lambda$runLater$10)(PlatformImpl.java:456)
    at java.base/java.security.AccessController.doPrivileged(AccessController.java:400)
    at [javafx.graphics@23/com.sun.javafx.application.PlatformImpl.lambda$runLater$11](mailto:javafx.graphics@23/com.sun.javafx.application.PlatformImpl.lambda$runLater$11)(PlatformImpl.java:455)
    at [javafx.graphics@23/com.sun.glass.ui.InvokeLaterDispatcher$Future.run](mailto:javafx.graphics@23/com.sun.glass.ui.InvokeLaterDispatcher$Future.run)(InvokeLaterDispatcher.java:95)
    at [javafx.graphics@23/com.sun.glass.ui.win.WinApplication._runLoop](mailto:javafx.graphics@23/com.sun.glass.ui.win.WinApplication._runLoop)(Native Method)
    at [javafx.graphics@23/com.sun.glass.ui.win.WinApplication.lambda$runLoop$3](mailto:javafx.graphics@23/com.sun.glass.ui.win.WinApplication.lambda$runLoop$3)(WinApplication.java:184)
    ... 1 more
```

I know that the binding is in a "stale/abandoned" state and I don't even know if the mighty Neil is still around or if it has some interest in this, for this reason I open this issue in the hope that someone will chime in and help me understanding how to help GStreamers to fix this.

In the current state the binding is unusable and next version of GStreamer will became unusable too.

I [opened an issue to GStreamer](https://gitlab.freedesktop.org/gstreamer/gstreamer/-/issues/4104) but they wants infos I'm not able to give.

If someone wants to join me in trying to troubleshoot this problem will be really appreciated.
It's really sad to see this great binding in a "non working state".

**Comment by matthiasblaesing** [comment]
This is not reproducible for me:

1. I downloaded https://gitlab.freedesktop.org/gstreamer/cerbero/-/jobs/68147502/artifacts/file/gstreamer-1.0-linux-x86_64-1.25.0.1.tar.xz
2. Expanded that
3. Set LD_LIBRARY_PATH to point to `<base>/gstreamer-1.0-linux-x86_64-1.25.0.1/lib/x86_64-linux-gnu/`
4. Set PATH to include `<base>/gstreamer-1.0-linux-x86_64-1.25.0.1/bin`
5. I build latest `gst1-java-core` from source
6. Ran `jshell --class-path <gst-checkout>/target/gst1-java-core-1.2.0-SNAPSHOT.jar:<home>/.m2/repository/net/java/dev/jna/jna/5.15.0/jna-5.15.0.jar`
7. Ran this:
    ```
    matthias@enterprise:~/src/gstreamer1.x-java$ jshell --class-path target/gst1-java-core-1.2.0-SNAPSHOT.jar:/home/matthias/.m2/repository/net/java/dev/jna/jna/5.15.0/jna-5.15.0.jar
    |  Willkommen bei JShell - Version 21.0.5
    |  Geben Sie für eine Einführung Folgendes ein: /help intro

    jshell> import org.freedesktop.gstreamer.Gst;

    jshell> import org.freedesktop.gstreamer.State;

    jshell> Gst.getVersionString()
    $3 ==> "GStreamer 1.25.0 (GIT)"

    jshell> Gst.init("dummy", "");
    $4 ==> String[1] { "" }

    jshell> var pipeline = Gst.parseLaunch("videotestsrc pattern=18 ! autovideosink")
    pipeline ==> Pipeline: [pipeline0]

    jshell> pipeline.setState(State.PLAYING);
    $6 ==> ASYNC

    jshell> 
    matthias@enterprise:~/src/gstreamer1.x-java$ 
    ```

I get the testvideo output. No crash.

BUT if I remember correctly JavaFX contains a subset of gstreamer and glib, maybe these clash? I suggest to create a minimal reproducer for the crash.

**Comment by sblantipodi** [comment]
Hi Matthias,
nice to see that you are still around the project :)

I spent a lot of time trying to debug this thing and at the end it was my IDE with some weird env variable set causing the problem.
I noticed the problem once I runned the build on my external CI Pipeline using another environment and it worked.

I feel a bit dumb but this is what it is.

Thank you for the kind answer and for spending time trying this thing, I'm sorry for the false alarm.
